src/cart_tumor.cc

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2025 compiler-research.org, Salvador de la Torre Gonzalez, Luciana
+ * Melina Luque
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *     SPDX-License-Identifier: Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file contains a model developed under Google Summer of Code (GSoC)
+ * for the compiler-research.org organization.
+ */
+
+#include "cart_tumor.h"
+#include "diffusion_thomas_algorithm.h"
+#include "forces_tumor_cart.h"
+#include "hyperparams.h"
+#include "tumor_cell.h"
+#include "utils_aux.h"
+#include "core/container/math_array.h"
+#include "core/diffusion/diffusion_grid.h"
+#include "core/environment/uniform_grid_environment.h"
+#include "core/model_initializer.h"
+#include "core/operation/mechanical_forces_op.h"
+#include "core/operation/operation.h"
+#include "core/param/param.h"
+#include "core/real_t.h"
+#include "core/simulation.h"
+#include <iostream>
+#include <memory>
+#include <vector>
+
+namespace bdm {
+
+int Simulate(int argc, const char** argv) {
+  // Set simulation bounds
+  auto set_param = [](Param* param) {
+    param->random_seed = kSeed;  // Set a fixed random seed for reproducibility
+    param->bound_space = Param::BoundSpaceMode::kTorus;  // Periodic boundary
+    param->min_bound =
+        -kBoundedSpaceLength /
+        2.0;  // NOLINT(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)
```

Comment:
Please uncomment these and let's see how to solve them if they are not good we can disable the checks.